cartography/intel/aws/identitycenter.py

```diff
@@ -0,0 +1,371 @@
+        aws_update_tag=aws_update_tag,
+    )
+
+
+@timeit
+def get_sso_users(boto3_session: boto3.session.Session, identity_store_id: str, region: str) -> List[Dict]:
+    """
+    Get all SSO users for a given Identity Store
+    """
+    client = boto3_session.client('identitystore', region_name=region)
+    users = []
+
+    try:
+        paginator = client.get_paginator('list_users')
+        for page in paginator.paginate(IdentityStoreId=identity_store_id):
+            user_page = page.get('Users', [])
+            for user in user_page:
+                if user.get('ExternalIds', None):
+                    user['ExternalId'] = user.get('ExternalIds')[0].get('Id')
+                users.append(user)
+    except client.exceptions.ClientError as e:
+        logger.warning(f"Failed to get SSO users for identity store {identity_store_id} in region {region}: {e}")
+        return []
+
+    return users
+
+
+@timeit
+def load_sso_users(
+    neo4j_session: neo4j.Session,
+    users: List[Dict],
+    identity_store_id: str,
+    region: str,
+    aws_update_tag: int,
+) -> None:
+    """
+    Load SSO users into the graph
+    """
+    logger.info(f"Loading {len(users)} SSO users for identity store {identity_store_id} in region {region}")
+
+    load(
+        neo4j_session,
+        SSOUserSchema(),
+        users,
+        lastupdated=aws_update_tag,
+        IdentityStoreId=identity_store_id,
+        Region=region,
+    )
+
+
+@timeit
+def get_role_assignments(
+    boto3_session: boto3.session.Session,
+    users: List[Dict],
+    instance_arn: str,
+    region: str,
+) -> List[Dict]:
+    """
+    Get role assignments for SSO users
+    """
+
+    logger.info(f"Getting role assignments for {len(users)} users")
+    client = boto3_session.client('sso-admin', region_name=region)
+    role_assignments = []
+
+    for user in users:
+        user_id = user['UserId']
+        try:
+            paginator = client.get_paginator('list_account_assignments_for_principal')
+            for page in paginator.paginate(InstanceArn=instance_arn, PrincipalId=user_id, PrincipalType='USER'):
+                for assignment in page.get('AccountAssignments', []):
+                    role_assignments.append({
+                        'UserId': user_id,
+                        'PermissionSetArn': assignment.get('PermissionSetArn'),
+                        'AccountId': assignment.get('AccountId'),
+                    })
+        except client.exceptions.ClientError as e:
+            logger.warning(f"Failed to get account assignments for user {user_id}: {e}")
+
+    return role_assignments
+
+
+@timeit
+def load_role_assignments(
+    neo4j_session: neo4j.Session,
+    role_assignments: List[Dict],
+    aws_update_tag: int,
+) -> None:
+    """
+    Load role assignments into the graph
+    """
+    logger.info(f"Loading {len(role_assignments)} role assignments")
+    if role_assignments:
+        neo4j_session.run(
```

Comment:
Let's use the data model here. I think we can do this by creating a `SSOUserToRole` relationship.

We can have a `transform` step that shapes the data returned from `get_sso_users()` and `get_role_assignments()` into what we need.

This way we don't have to use `neo4j_session.run()` (since this can fail and does not retry automatically), and the data model does batching and automatic cleanups and all that.

Let's get on a call if you're stuck with this. I'm also happy to push changes up to your branch if that's easier.